src/util/symbol.h

```diff
@@ -68,6 +68,10 @@ class symbolt
   bool is_lvalue, is_file_local, is_extern, is_volatile,
        is_parameter, is_auxiliary, is_weak;
 
+  // Can a static variable be nondet-initialized? User static variables can be
+  // but proprietary ones no (e.g., inflight_exception, clinit_already_run etc)
+  bool is_static_nondet_init = false;
```

Comment:
This new field needs to be serialised and de-serialiased when writing/reading goto binaries. Also text output needs to be updated.

Reply:
Why not just `is_nondet_init` (or some more grammatically correct name)? (The `static` information is already in `is_static_lifetime` if needed).